blotztask-ui/src/app/dashboard/navbar/categories.tsx

```diff
@@ -1,23 +1,40 @@
 import React from "react";
+import { useEffect, useState } from 'react';
+import { LabelDTO } from '@/model/label-dto';
+import { fetchAllLabel } from '@/services/labelService';
 
 export function Categories() {
-  const categories = [
-    { name: "Personal", color: "bg-yellow-500" },
-    { name: "Academic", color: "bg-cyan-500" },
-    { name: "Others", color: "bg-blue-500" },
-    { name: "Work", color: "bg-purple-500" },
-  ];
+  const [labels, setLabels] = useState<LabelDTO[]>([]);
+
+  const loadAllLabel = async () => {
+      try {
+        const labelData = await fetchAllLabel();
+        setLabels(labelData);
+      } catch (error) {
+        console.error('Error loading labels:', error);
+      }
+    };
+  
+    useEffect(() => {
+      loadAllLabel();
+    }, []);
```

Comment:
```suggestion
  const loadAllLabel = async () => {
      try {
        const labelData = await fetchAllLabel();
        setLabels(labelData);
      } catch (error) {
        console.error('Error loading labels:', error);
      }
    };
  
    useEffect(() => {
      loadAllLabel();
    }, []);
```

This all is consider as data fetching logic , we should move this to parent component (smart component)

read about why we do this as a good practices - https://medium.com/@thejasonfile/dumb-components-and-smart-components-e7b33a698d43

Reply:
![image](https://github.com/user-attachments/assets/8b3a3696-277e-47ee-91f1-c4c2703d09ba)
Moved data fetching logic to parent component